pkg/dockerregistry/server/auth.go

```diff
@@ -343,12 +344,36 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 				}
 				verifiedPrune = true
 			default:
+				if repoAccessVerified {
```

Comment:
The bool is not enough. A set of all `<namespace>/<isname>` needs to be checked and updated.